pdns/dnsdist.cc

```diff
@@ -2390,6 +2390,32 @@ static void setupLocalSocket(ClientState& clientState, const ComboAddress& addr,
         warnlog(e.what());
       }
     }
+#ifdef __linux__
+#ifdef SO_RCVBUF
+    {
+      std::ifstream ifs("/proc/sys/net/core/rmem_max");
+      if (ifs) {
+        std::string line;
+        if (getline(ifs, line)) {
+          auto maxRmem = std::stoi(line);
+          SSetsockopt(socket, SOL_SOCKET, SO_RCVBUF, maxRmem);
+        }
+      }
+    }
+#endif /* SO_RCVBUF */
+#ifdef SO_SNDBUF
+    {
+      std::ifstream ifs("/proc/sys/net/core/wmem_max");
+      if (ifs) {
+        std::string line;
+        if (getline(ifs, line)) {
+          auto maxWmem = std::stoi(line);
+          SSetsockopt(socket, SOL_SOCKET, SO_SNDBUF, maxWmem);
+        }
```

Comment:
Perhaps log at info level after raising the buffer size, advertising the new value?